pkgs/misc/vim-plugins/overrides.nix

```diff
@@ -108,6 +109,22 @@ self: super: {
     '';
   });
 
+
+  coc-nvim = let
+    version = "v0.0.67";
+    index_js = fetchzip {
+        url = "https://github.com/neoclide/coc.nvim/releases/download/${version}/coc.tar.gz";
+        sha256 = "0cqgrfyaq9nck1y6mb63gmwgdrxqzgdgns5gjshpp1xzfq6asrqj";
+      };
+  in super.coc-nvim.overrideAttrs(old: {
+    # you still need to enable the node js provider in your nvim config
```

Comment:
Any way to notify the user of this at runtime? Could be confusing otherwise.

Reply:
Would be happy to know but I can't think of any. Ideally the vim infrastructure could enrich the vimrc config. Might be possible after my neovim PR but that will take ages and coc.nvim is really really good so even if it can only be managed imperatively it's a fundation :p

Reply:
Couldn't we patch coc.nvim?

My fear is that users try to install it and then lose a lot of time debugging before they find that comment.

Reply:
Maybe one could patch coc.nvim g:coc_node_path but I don't think we should as it's best to use the nodejs provider in neovim (I don't know how it works on vim, maybe they just set g:coc_node_path): it means setting neovim with `withNodeJs`. 
Anyway you can't get anything with this kind of plugins out of the box, so anyway the users will have to configure them. I would rather spend time on https://github.com/NixOS/nixpkgs/pull/55635

Reply:
I mean it feels hackish while solving very little (and maybe add confusion as to why the neovim node provider isn't used). It's not possible to provide an out of the box experience for coc.nvim. If you want I can add some documentation on the wiki.

Reply:
I thought more in the direction of giving a useful error message when the user hasn't made the appropriate configuration changes.

> Anyway you can't get anything with this kind of plugins out of the box

So this is the case with e.g. vim-plug too, not just a nix issue?

Anyway, I don't want to block  this. Feel free to merge if you think this is fine as is.

Reply:
I wish I could enforce the config but it's not possible yet so let's merge.
You can run `:checkhealth` for warnings.
It's an LSP plugin, actually the most popular so user have to configure the language servers anyway/possibly install yarn to get extensions so it's involving see https://github.com/NixOS/nixpkgs/pull/56091 . Rather than having it sit for too long, let's start small and improve things little by little.